### DISCUSSION TITLE
Navigation withour pushing.

### DISCUSSION THREAD

**Discussion by spacekestrel** [jonataslaw/getx#1348] [Q&A]
Is there any way to implement such behavior, when on event instead of pushing to new page getx just change url? 
For ex. from somewebsite.com/control_panel to  somewebsite.com/control_panel#payment_methods

**Comment by zambetpentru** [comment]
Hi, this would be a very helpful feature to have - e.g. in a document view to show ?pageno=32 and have that update as people move through the document and for sharing. 

If you have any workarounds / pointers on places to look please, I found this but didn't get any further yet.

` /// On Web, updates the URL bar with the [routeName].
  static void routeUpdated({
    String? routeName,
    String? previousRouteName
  }) {
    SystemChannels.navigation.invokeMethod<void>(
      'routeUpdated',
      <String, dynamic>{
        'previousRouteName': previousRouteName,
        'routeName': routeName,
      },
    );
  }`
  
  This also 'works' but gets overwritten again immediately:

` 
import 'dart:html' as html;
html.window.location.href = 'new URL'`

**Reply by spacekestrel** [reply]
> 
> 
> Hi, this would be a very helpful feature to have - e.g. in a document view to show ?pageno=32 and have that update as people move through the document and for sharing.
> 
> If you have any workarounds / pointers on places to look please, I found this but didn't get any further yet.
> 
> ` /// On Web, updates the URL bar with the [routeName]. static void routeUpdated({ String? routeName, String? previousRouteName }) { SystemChannels.navigation.invokeMethod<void>( 'routeUpdated', <String, dynamic>{ 'previousRouteName': previousRouteName, 'routeName': routeName, }, ); }`
> 
> This also 'works' but gets overwritten again immediately:
> 
> ` import 'dart:html' as html; html.window.location.href = 'new URL'`

yea, this is not very good implementation. I guess, it's better to stick with react or vue.
